cmd/cape/cmd/run.go

```diff
@@ -77,14 +77,37 @@ func run(cmd *cobra.Command, args []string) error {
 	insecure := C.Insecure
 
 	if len(args) < 1 {
-		return UserError{Msg: "you must pass a function ID", Err: fmt.Errorf("invalid number of input arguments")}
+		return UserError{Msg: "you must pass a function ID or a function name", Err: fmt.Errorf("invalid number of input arguments")}
 	}
 
 	if len(args) > 2 {
 		return UserError{Msg: "you must pass in only one input data (stdin, string or filename)", Err: fmt.Errorf("invalid number of input arguments")}
 	}
 
-	functionID := args[0]
+	t, err := getAuthToken()
+	if err != nil {
+		return err
+	}
+	auth := entities.FunctionAuth{Type: entities.AuthenticationTypeAuth0, Token: t}
+
+	functionToken, _ := cmd.Flags().GetString("token")
+	if functionToken != "" {
+		auth.Type = entities.AuthenticationTypeFunctionToken
+		auth.Token = functionToken
+	}
+
+	function := args[0]
+	functionID := function
+	if strings.Contains(function, "/") {
```

Comment:
perhaps we should do some validation + friendly error messages here? E.g., `cape run cool/` shouldn't work but it would pass this check

Reply:
I was thinking instead of introducing new error messages for validation here, use the validation errors bubbled up by the supervisor(also to keep the errors consistent across all sdks), so in function_id.go in the sdk, something like
```
type supervisorErrorMsg struct {
	message string `json:"message"`
}

func GetFunctionID(functionReq FunctionIDRequest) (string, error) {
	// .... Send request

	if res.StatusCode != http.StatusOK {
		// return the error message from supervisor
		var e supervisorErrorMsg
		err = json.Unmarshal(body, &e)
		return "", fmt.Errorf("HTTP Error: %d\n %s", res.StatusCode, e.message)
	}
```
Would that work?

Reply:
Added some validation tests for the cli